src/SourceBuild/content/prep.sh

```diff
@@ -67,57 +68,76 @@ done
 
 # Attempting to bootstrap without an SDK will fail. So either the --no-sdk flag must be passed
 # or a pre-existing .dotnet SDK directory must exist.
-if [[ "$buildBootstrap" == "true" && "$installDotnet" == "false" && ! -d $SCRIPT_ROOT/.dotnet ]]; then
+if [ "$buildBootstrap" == true ] && [ "$installDotnet" == false ] && [ ! -d "$SCRIPT_ROOT/.dotnet" ]; then
     echo "  ERROR: --no-sdk requires --no-bootstrap or a pre-existing .dotnet SDK directory.  Exiting..."
-    exit -1
+    exit 1
 fi
 
 # Check to make sure curl exists to download the archive files
 if ! command -v curl &> /dev/null
 then
     echo "  ERROR: curl not found.  Exiting..."
-    exit -1
+    exit 1
+fi
+
+GIT_DIR="$SCRIPT_ROOT/.git"
+
+if [ -f "$GIT_DIR/config" ]; then
+  if [ -n "$sourceUrl" ] || [ -n "$sourceVersion" ]; then
+    echo "ERROR: $SCRIPT_ROOT is a git repository, --source-repository and --source-version cannot be used."
```

Comment:
Curious why this condition is needed?  It feels like what is specified via the options should win.  If you passed in the options, it will create the git files.  It appears that it prevents you try running the script a second time it will fail.  That doesn't feel right.

Reply:
Fair point, `prep.sh` should be idempotent. Let's address this once we decide whether creating the `.git` files is the right way or we should try and fix this somewhere down the line by passing the info through MSBuild somehow.